### PR TITLE
Differentiate the username/password for pulling base image with Jib

### LINKED ISSUE #8498: Jib pull base image from public registry and push to private registry
**Describe the bug**
Building an image with the container-jib extension fails, if the base image is hosted in a public registry and the properties `quarkus.container-image.username` and `quarkus.container-image.password` are set to push on a different private registry.

This is a regression from https://github.com/quarkusio/quarkus/pull/8288.

**Expected behavior**
Build image from the public registry, and push the image to a different private registry.

**Actual behavior**
```
20695 [ERROR] Caused by: com.google.cloud.tools.jib.api.RegistryAuthenticationFailedException: Failed to authenticate with registry registry-1.docker.io/azul/zulu-openjdk because: 401 Unauthorized
20695 [ERROR] {"details":"incorrect username or password"}
20695 [ERROR] 
20695 [ERROR] 	at com.google.cloud.tools.jib.registry.RegistryAuthenticator.authenticate(RegistryAuthenticator.java:306)
20695 [ERROR] 	at com.google.cloud.tools.jib.registry.RegistryAuthenticator.authenticate(RegistryAuthenticator.java:258)
20696 [ERROR] 	at com.google.cloud.tools.jib.registry.RegistryAuthenticator.authenticatePull(RegistryAuthenticator.java:177)
20696 [ERROR] 	at com.google.cloud.tools.jib.registry.RegistryClient.doBearerAuth(RegistryClient.java:362)
20696 [ERROR] 	at com.google.cloud.tools.jib.registry.RegistryClient.doPullBearerAuth(RegistryClient.java:327)
20696 [ERROR] 	at com.google.cloud.tools.jib.builder.steps.PullBaseImageStep.call(PullBaseImageStep.java:158)
20696 [ERROR] 	at com.google.cloud.tools.jib.builder.steps.PullBaseImageStep.call(PullBaseImageStep.java:56)
20697 [ERROR] 	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
20697 [ERROR] 	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
20697 [ERROR] 	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
20697 [ERROR] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
20697 [ERROR] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
20698 [ERROR] 	at java.base/java.lang.Thread.run(Thread.java:832)
```

**To Reproduce**
1. Set `quarkus.jib.base-jvm-image=docker.io/azul/zulu-openjdk:14`
2. Set `quarkus.container-image.registry=registry.gitlab.com`, `quarkus.container-image.group` and `quarkus.container-image.name`.
3. `./mvnw package -Dquarkus.container-image.push=true -Dquarkus.container-image.username=REGISTRY_USER -Dquarkus.container-image.password=REGISTRY_PASSWORD`

**Configuration**
```properties
quarkus.jib.base-jvm-image=docker.io/azul/zulu-openjdk:14
quarkus.container-image.registry=registry.gitlab.com
quarkus.container-image.group=org/group
quarkus.container-image.name=project
```

**Environment (please complete the following information):**
 - Output of `java -version`: openjdk version "14" 2020-03-17
 - Quarkus version or git rev: 1.3.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3

**Additional context**
The public base image is set to `docker.io` explicitly (it's the default if not set), the `quarkus.container-image.registry` is set to `registry.gitlab.com`, since both registries are different one approach is to detect if they are the same to authenticate using the container-image properties. 

Thinking more on this, if both registries are different (both private) the best would be to provide another set of permissions (user/pass) to be used by Jib to pull and push.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #8498

### CODE REVIEW

**Comment by geoand** [comment]
We don't do the same thing for Docker because that would involve parsing the `Dockerfile` to figure out the registry of the base image.
It's possible, but I would like to avoid parsing the Dockerfile unless we really see users needs this feature.

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java [hunk 1]
```diff
@@ -44,5 +45,17 @@
...
+     * The username to use to authenticate with the registry used to pull the base JVM image
+     */
+    @ConfigItem
+    public Optional<String> baseRegistryUsername;
+
+    /**
+     * The password to use to authenticate with the registry used to pull the base JVM image
+     */
+    @ConfigItem
+    public Optional<String> baseRegistryPassword;
```
Comment (gastaldi) [inline extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:60]: Why the `base` prefix here? Just curious

Comment (geoand) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:60]: It's not great for sure, but I added it to keep consistency with the `baseJvmImage` and `baseNativeImage` properties.
Does that sound good?

Comment (gastaldi) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:60]: I'd just go with `registryUsername` and `registryPassword` then. The `base` in this case doesn't make much sense, I think

**Review by gastaldi — APPROVED** [review]

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java (hunk 1 above)
Comment (gastaldi) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:60]: Nevermind my previous comment, I see there is an username and password in the container-image config

Comment (geoand) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:60]: Exactly :)

**Comment by gastaldi** [comment]
Looks like you need to change the commit message to the correct issue?

**Comment by geoand** [comment]
Yeah, you are regiht

**Pull request merged**